daemon/algod/api/server/v2/handlers.go

```diff
@@ -1670,3 +1672,26 @@ func (v2 *Handlers) TealDisassemble(ctx echo.Context) error {
 func (v2 *Handlers) ExperimentalCheck(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, true)
 }
+
+// GetBlockTimeStampOffset gets the timestamp offset.
+// This is only available in dev mode.
+// (GET /v2/devmode/blocks/offset)
+func (v2 *Handlers) GetBlockTimeStampOffset(ctx echo.Context) error {
+	offset := v2.Node.GetBlockTimeStampOffset()
+	if offset == nil {
+		err := fmt.Errorf("cannot get timestamp offset because we are not in dev mode")
+		return badRequest(ctx, err, fmt.Sprintf(errFailedRetrievingTimeStampOffset, err), v2.Log)
+	}
+	return ctx.JSON(http.StatusOK, model.GetBlockTimeStampOffsetResponse{Offset: uint64(*offset)})
+}
+
+// SetBlockTimeStampOffset sets the timestamp offset.
+// This is only available in dev mode.
+// (POST /v2/devmode/blocks/offset/{offset})
+func (v2 *Handlers) SetBlockTimeStampOffset(ctx echo.Context, offset uint64) error {
+	err := v2.Node.SetBlockTimeStampOffset(int64(offset))
```

Comment:
offset should be uint64?

Reply:
I casted it here because I think the OpenAPI specs demand arguments to be `uint64` while timestamps are in `int64`.

Reply:
we want to check for max offset value here, and internally set offset value to max int64 when necessary. 
if int overflow occurs, offset may be negative. 
https://go.dev/play/p/ZSfCiwMjQoe

Reply:
That's a good point! I added a check to return a 400 if the value exceeds the int64 bounds and added a check: [430ff14](https://github.com/algorand/go-algorand/pull/5296/commits/430ff146dd11eca32859657a02723826679805b1)

Reply:
now I wonder what happens when the block timestamp overflows. Guess I can try it out now :joy:

Reply:
I will check for this edge case when setting the timestamp 😅  [0dc2458](https://github.com/algorand/go-algorand/pull/5296/commits/0dc2458e73b8941ba897539f69b48eda286b0cbf)